modelforge/dataset/parameters.py

```diff
@@ -0,0 +1,100 @@
+from enum import Enum
+from typing import Optional, List, Dict
+from pydantic import BaseModel, ConfigDict, Field, model_validator
+from typing_extensions import Self
+
+
+class CaseInsensitiveEnum(str, Enum):
```

Comment:
What does enum do?

Reply:
I defined `CaseInsensitiveEnum` to save effort later ( I added in a method that allows us to ignore string capitalization during assignment), and then can just subclass uses this as the parent .  More broadly speaking, setting up a class that inherits from Enum is the way to specify "choices" in a pydantic model, basically limiting what values can be accepted as input (it will produced a validation error if you give something not listed in the enum).